trace_replay/io_tracer.cc

```diff
@@ -216,13 +216,21 @@ void IOTracer::EndIOTrace() {
 }
 
 Status IOTracer::WriteIOOp(const IOTraceRecord& record) {
+  Status s;
```

Comment:
I am not sure I understand the need for any of the changes in this method.  This method looks like it should have passed the ASC without these changes.  Can you revert them and tell me what test is failing?

Reply:
io_tracer_parser_test fails because currently, I am ignoring the status of IOTracer::WriteIOOp in file_system_tracer.cc. I am planning to either log or return that status but haven't figured it out what to do with status. I added TODO as well. Planning to work on this in next half with few more additions related to io_tracer.

Reply:
interesting. I wonder if this approach relies on return-value optimization. I'd expect if a different `Status` were constructed in the client with the result, that one would be unchecked.

Reply:
```
$ make clean && ASSERT_STATUS_CHECKED=1 EXTRA_CXXFLAGS="-fno-elide-constructors" make -j48 io_tracer_parser_test
$ ./io_tracer_parser_test
[==========] Running 4 tests from 1 test case.                                                                                                                                                                                        [2/1853]
[----------] Global test environment set-up.
[----------] 4 tests from IOTracerParserTest
[ RUN      ] IOTracerParserTest.InvalidArguments
Failed to check Status 0x7ffe392aa5b0
#0   ./io_tracer_parser_test() [0x69fc3f] rocksdb::port::PrintStack(int)        /data/users/andrewkr/rocksdb/port/stack_trace.cc:121
#1   ./io_tracer_parser_test() [0x470a39] rocksdb::Status::~Status()    /data/users/andrewkr/rocksdb/./include/rocksdb/status.h:42
#2   ./io_tracer_parser_test() [0x510135] rocksdb::SanitizeOptions(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rocksdb::DBOptions const&)   /data/users/andrewkr/rocksdb/db/db_impl/db_impl_open.cc:166 (discriminator 1)
#3   ./io_tracer_parser_test() [0x4c4392] rocksdb::DBImpl::DBImpl(rocksdb::DBOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)        /data/users/andrewkr/rocksdb/db/db_impl/db_impl.cc:153
#4   ./io_tracer_parser_test() [0x512b0e] rocksdb::DBImpl::Open(rocksdb::DBOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<rocksdb::ColumnFamilyDescriptor, std::allocator<rocksdb::ColumnFamilyDescriptor> > const&, std::vector<ro      ksdb::ColumnFamilyHandle*, std::allocator<rocksdb::ColumnFamilyHandle*> >*, rocksdb::DB**, bool, bool)  /data/users/andrewkr/rocksdb/db/db_impl/db_impl_open.cc:1519
#5   ./io_tracer_parser_test() [0x514a98] rocksdb::DB::Open(rocksdb::DBOptions const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<rocksdb::ColumnFamilyDescriptor, std::allocator<rocksdb::ColumnFamilyDescriptor> > const&, std::vector<rocksd      ::ColumnFamilyHandle*, std::allocator<rocksdb::ColumnFamilyHandle*> >*, rocksdb::DB**)  /data/users/andrewkr/rocksdb/db/db_impl/db_impl_open.cc:1450
#6   ./io_tracer_parser_test() [0x514c98] rocksdb::DB::Open(rocksdb::Options const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, rocksdb::DB**)     /data/users/andrewkr/rocksdb/db/db_impl/db_impl_open.cc:1428
#7   ./io_tracer_parser_test() [0x4746a7] rocksdb::IOTracerParserTest::IOTracerParserTest()     /data/users/andrewkr/rocksdb/tools/io_tracer_parser_test.cc:42
#8   ./io_tracer_parser_test() [0x47599b] rocksdb::IOTracerParserTest_InvalidArguments_Test::IOTracerParserTest_InvalidArguments_Test() /data/users/andrewkr/rocksdb/tools/io_tracer_parser_test.cc:108
#9   ./io_tracer_parser_test() [0x4a4e3d] testing::Test* testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::TestFactoryBase, testing::Test*>(testing::internal::TestFactoryBase*, testing::Test* (testing::internal::TestFactoryBase::*)(), char const*)    /data/users/andrewkr/rocksdb/third-party/gtest-1.8.1/fused-src/gtest/gtest-all.cc:3899
...
```

Reply:
Actually the above failure I showed is indeed a different optimization-dependent `Status` check. I guess there is a whole project to be done to get rid of optimization dependencies. Please feel free to go ahead then and not block on this.

Reply:
@ajkr I will be submitting a separate PR shortly that addresses the "-fno-elide-constructors".